package maidsafe.vault.protobuf;

message MetadataValue {
  required int32 size = 1;
  optional int64 subscribers = 2;
  repeated bytes online_pmid_name = 3;
  repeated bytes offline_pmid_name = 4;
}

message DataOrProof {
  message Data {
    required int32 type = 1;
    required bytes name = 2;
    required bytes content = 3;
  }
  message Proof {
    required int32 type = 1;
    required bytes name = 2;
    required bytes value = 3;
    required int32 size = 4;
  }
  required bytes node_id = 1;
  optional bytes serialised_data = 2;
  required bytes serialised_proof = 3;
  required bytes signature = 4;
}

